src/pages/views/PageDetails.tsx

```diff
@@ -59,22 +74,99 @@ export const PageDetails: React.FC<PageDetailsProps> = ({ id, params }) => {
       {(pageRemove, pageRemoveOpts) => (
         <TypedPageUpdate>
           {(pageUpdate, pageUpdateOpts) => (
             <TypedPageDetailsQuery variables={{ id }}>
               {pageDetails => {
-                const handleUpdate = async (data: PageData) => {
+                const handleUpdate = async (data: PageSubmitData) => {
+                  let errors: Array<
+                    | AttributeErrorFragment
+                    | UploadErrorFragment
+                    | PageErrorFragment
+                  > = [];
+
+                  const uploadFilesResult = await Promise.all(
+                    data.attributesWithNewFileValue.map(fileAttribute =>
+                      uploadFile({
+                        variables: {
+                          file: fileAttribute.value
+                        }
+                      })
+                    )
+                  );
+
+                  const attributesWithAddedNewFiles: AttributeValueInput[] = uploadFilesResult.reduce(
+                    (attributesWithAddedFiles, uploadFileResult, index) => {
+                      const attribute = data.attributesWithNewFileValue[index];
+
+                      errors = [
+                        ...errors,
+                        ...uploadFileResult.data.fileUpload.uploadErrors
+                      ];
+                      return [
+                        ...attributesWithAddedFiles,
+                        {
+                          file:
+                            uploadFileResult.data.fileUpload.uploadedFile.url,
+                          id: attribute.id,
+                          values: []
+                        }
+                      ];
+                    },
+                    []
+                  );
+
                   const result = await pageUpdate({
                     variables: {
                       id,
-                      input: createPageInput(data)
+                      input: createPageInput(data, attributesWithAddedNewFiles)
                     }
                   });
 
-                  return result.data.pageUpdate.errors;
+                  errors = [...errors, ...result.data.pageUpdate.errors];
+
+                  if (errors.length === 0) {
+                    const deleteAttributeValuesResult = await Promise.all(
+                      pageDetails?.data?.page?.attributes.map(
+                        existingAttribute => {
+                          const fileValueUnused =
+                            existingAttribute.attribute.inputType ===
+                              AttributeInputTypeEnum.FILE &&
+                            existingAttribute.values.length > 0 &&
+                            data.attributes.find(
+                              dataAttribute =>
+                                dataAttribute.id ===
+                                existingAttribute.attribute.id
+                            ).value.length === 0;
```

Comment:
Candidate for util function. Try to separate logic from components body.
Also, please avoid such long conditions. There is no way to comprehend whats its doing at first sight.

```suggestion
// rename data to something which describe what kind of data it is
const fileValueUnused = (data, existingAttribute) =>{
  if(existingAttribute.attribute.inputType !== AttributeInputTypeEnum.FILE){
    return false
  }
  if(existingAttribute.values.length === 0 ){
    return false
  }
  
  // did I got it right? Youre looking for modified attributes in "data"?
  const modifiedAttribute = data.attributes.find(
    dataAttribute =>
      dataAttribute.id ===
      existingAttribute.attribute.id
  )
  return modifiedAttribute.value.length === 0;
}
```